src/Flowdock/FlowdockHandler.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MonologHttp\Flowdock;
+
+use Monolog\Formatter\FormatterInterface;
+use Monolog\Logger;
+use MonologHttp\AbstractHttpClientHandler;
+use MonologHttp\Flowdock\Formatter\FlowdockFormatterInterface;
+use MonologHttp\Flowdock\Formatter\FlowdockMessageFormatter;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
+use Monolog\Handler\HandlerInterface;
+
+final class FlowdockHandler extends AbstractHttpClientHandler
+{
+
+    /**
+     * @var UriInterface|string
+     */
+    private $uri;
+
+    /**
+     * @var string|null
+     */
+    private $flowToken;
+
+    /**
+     * @param string|UriInterface $uri
+     * @param int|string $level
+     */
+    public function __construct(
+        ClientInterface $client,
+        RequestFactoryInterface $requestFactory,
+        $uri,
+        ?string $flowToken = null,
+        $level = Logger::ERROR,
+        bool $bubble = true
+    ) {
+        parent::__construct($client, $requestFactory, $level, $bubble);
+        $this->uri = $uri;
+        $this->flowToken = $flowToken;
+    }
+
+    protected function createRequest(array $record): RequestInterface
+    {
+        $body = \json_encode($record['formatted']);
+        if (\JSON_ERROR_NONE !== \json_last_error()) {
+            throw new \InvalidArgumentException('Could not format record to json: ' . \json_last_error_msg());
+        }
+
+        $url = (null === $this->flowToken) ?
+            $this->uri :
+            \sprintf('%s?flow_token=%s', $this->uri, $this->flowToken);
```

Comment:
maybe it's best if we could add it as a header after all... 

if the user passes a URL like `https://flowdock.com/test?whatever#my` this wont work...

also since we have strict types enabled this wont work if it is a `UriInterface`...

In order to avoid tricky hacks.. I suggest to move to header.

Reply:
The other way for the 'Source Token Authentication' is parse the token as a post parameter and hence parse to the formatter

Let me know if you have any thoughts on this